src/main/java/uk/gov/companieshouse/exemptions/AuthenticationFilter.java

```diff
@@ -2,32 +2,34 @@
 
 import java.io.IOException;
 import java.util.Optional;
-import javax.servlet.FilterChain;
-import javax.servlet.ServletException;
-import javax.servlet.http.HttpServletRequest;
-import javax.servlet.http.HttpServletResponse;
+import jakarta.servlet.FilterChain;
+import jakarta.servlet.ServletException;
+import jakarta.servlet.http.HttpServletRequest;
+import jakarta.servlet.http.HttpServletResponse;
 
 import org.apache.commons.lang.StringUtils;
 import org.apache.commons.lang3.ArrayUtils;
+import org.springframework.lang.NonNull;
 import org.springframework.web.filter.OncePerRequestFilter;
 import uk.gov.companieshouse.logging.Logger;
 
 public class AuthenticationFilter extends OncePerRequestFilter {
 
-    private final Logger logger;
+    private final Logger authLogger;
 
     public AuthenticationFilter(Logger logger) {
-        this.logger = logger;
+        this.authLogger = logger;
     }
 
     @Override
-    protected void doFilterInternal(HttpServletRequest request, HttpServletResponse response,
-                                    FilterChain filterChain) throws ServletException, IOException {
+    protected void doFilterInternal(HttpServletRequest request,
+                                    @NonNull HttpServletResponse response,
+                                    @NonNull FilterChain filterChain) throws ServletException, IOException {
```

Comment:
Is addition of NonNull necessary as part of upgrade or done as a good practice? If done as a good practice, would it be good to add it consistently in all places(where it makes sense) rather than only in some places?

Reply:
Not necessary as part of the upgrade but the IDE/sonarLint gave errors for only these parameters so added, hence why only in some places